packages/edit-site/src/components/global-styles/context-menu.js

```diff
@@ -24,48 +41,97 @@ function ContextMenu( { name, parentMenu = '' } ) {
 	const hasLayoutPanel = hasDimensionsPanel;
 	const hasVariationsPanel = useHasVariationsPanel( name, parentMenu );
 
+	const { canEditCSS } = useSelect( ( select ) => {
+		const { getEntityRecord, __experimentalGetCurrentGlobalStylesId } =
+			select( coreStore );
+
+		const globalStylesId = __experimentalGetCurrentGlobalStylesId();
+		const globalStyles = globalStylesId
+			? getEntityRecord( 'root', 'globalStyles', globalStylesId )
+			: undefined;
+
+		return {
+			canEditCSS:
+				!! globalStyles?._links?.[ 'wp:action-edit-css' ] ?? false,
+		};
+	}, [] );
+
 	return (
-		<ItemGroup>
-			{ hasTypographyPanel && (
-				<NavigationButtonAsItem
-					icon={ typography }
-					path={ parentMenu + '/typography' }
-					aria-label={ __( 'Typography styles' ) }
-				>
-					{ __( 'Typography' ) }
-				</NavigationButtonAsItem>
-			) }
-			{ hasColorPanel && (
-				<NavigationButtonAsItem
-					icon={ color }
-					path={ parentMenu + '/colors' }
-					aria-label={ __( 'Colors styles' ) }
-				>
-					{ __( 'Colors' ) }
-				</NavigationButtonAsItem>
-			) }
-			{ hasBorderPanel && (
-				<NavigationButtonAsItem
-					icon={ border }
-					path={ parentMenu + '/border' }
-					aria-label={ __( 'Border styles' ) }
-				>
-					{ __( 'Border' ) }
-				</NavigationButtonAsItem>
-			) }
-			{ hasLayoutPanel && (
-				<NavigationButtonAsItem
-					icon={ layout }
-					path={ parentMenu + '/layout' }
-					aria-label={ __( 'Layout styles' ) }
-				>
-					{ __( 'Layout' ) }
-				</NavigationButtonAsItem>
-			) }
-			{ hasVariationsPanel && (
-				<ScreenVariations name={ name } path={ parentMenu } />
-			) }
-		</ItemGroup>
+		<>
+			<ItemGroup>
+				{ hasTypographyPanel && (
+					<NavigationButtonAsItem
+						icon={ typography }
+						path={ parentMenu + '/typography' }
+						aria-label={ __( 'Typography styles' ) }
+					>
+						{ __( 'Typography' ) }
+					</NavigationButtonAsItem>
+				) }
+				{ hasColorPanel && (
+					<NavigationButtonAsItem
+						icon={ color }
+						path={ parentMenu + '/colors' }
+						aria-label={ __( 'Colors styles' ) }
+					>
+						{ __( 'Colors' ) }
+					</NavigationButtonAsItem>
+				) }
+				{ hasBorderPanel && (
+					<NavigationButtonAsItem
+						icon={ border }
+						path={ parentMenu + '/border' }
+						aria-label={ __( 'Border styles' ) }
+					>
+						{ __( 'Border' ) }
+					</NavigationButtonAsItem>
+				) }
+				{ hasLayoutPanel && (
+					<NavigationButtonAsItem
+						icon={ layout }
+						path={ parentMenu + '/layout' }
+						aria-label={ __( 'Layout styles' ) }
+					>
+						{ __( 'Layout' ) }
+					</NavigationButtonAsItem>
+				) }
+				{ hasVariationsPanel && (
+					<ScreenVariations name={ name } path={ parentMenu } />
+				) }
+				{ !! parentMenu && !! canEditCSS && (
```

Comment:
With this condition, the custom CSS panel shows when you navigate to Styles > Blocks > Block but also when the block's style variation panel is opened ( Styles > Blocks > Block > Style Variation ).
You can test this by selecting the image block or any other block that has a custom block style.
How can I prevent the CSS panel from displaying in the style variation panel?

@Mamaduka @tellthemachines

Reply:
@carolinan, this is on my to-do list. I will try to get back to you with the answer ASAP.

Reply:
```suggestion
				{  isBlocksPanel && canEditCSS && (
```

I think we can use `parentMenu` to determine the current panel. The variations panel [uses a similar check](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/variations-panel.js#L19).

```js
const isBlocksPanel =
		parentMenu.includes( 'blocks' ) &&
		! parentMenu.includes( 'variations' );
```

P.S. There's no need for a double exclamation before `canEditCSS` it's already a boolean value.